### DISCUSSION TITLE
Local IP to public IP

### DISCUSSION THREAD

**Discussion by makhlwf** [termux/termux-app#3591] [Q&A] [answered]
**Hello everyone,**

I have completed a simple website with a friend, and we have both hosted it on our phones using Termux with the Apache dependency. The website is now successfully running on the local IP address 192.168.1.4, which is the local Wi-Fi network IP. However, we want to make our websites accessible via the public IP address like 165.12.3.4, so that both of us, living in different countries, can view our websites.

I consulted artificial intelligence about this issue, and it informed me that I need to set up port triggering for 192.168.1.4 in the Wi-Fi network settings to route incoming connections through 165.12.3.4. This is what I understood.

When I did that, my friend and I both tried to access our respective IPs, but in the browser, it shows "connection time out - IP took a long time to open." 

Does anyone know a solution to this problem?

Thank you.

**Comment by TomJo2000** [comment, marked as answer]
> I consulted artificial intelligence about this issue, [...]

That was your first mistake right there.

What you are looking for is either **Port Forwarding**, or a **Reverse Proxy**.
Port Triggering isn't completely wrong, and may even work for your usecase, but it is mainly used for IRC clients.

---

To illustrate the issue more clearly.
Your *local* HTTP server on the *local* IPv4 address `192.168.1.4` is serving your website on, presumably Port 80 (HTTP) and/or Port 443 (HTTPS).

Those local ports are ~~not~~ exposed on the LAN, so you should be able to access the website from any device on the same network.

If you want to access them from the internet you will need to setup forwarding for Port 80 (and/or 443) from your local device.
Meaning, your router will "expose" Port 80 (and/or 443) and direct any traffic directed to <your public IP>:80 to the HTTP server running on your local host.

This can either be done through the use of a reverse proxy which will receive requests from the internet and forward the connection to the appropriate server based on the request content (protocol, requested URL, etc).
(The reverse proxy itself will need to be exposed to the internet obviously)

Or more simply by just forwarding the relevant Port(s) from your "server" directly.
(Port forwarding is usually configured at the router)

**Comment by twaik** [comment]
AFAIK this can not be implemented and discussion can be closed.

**Comment by TomJo2000** [comment]
Put another way.
Termux isn't responsible for, and can't by itself, set up Port forwarding.
